iTunesSearch/iTunesSearch/SearchResultController.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  SearchResultController.swift
+//  iTunesSearch
+//
+//  Created by Dojo on 7/8/20.
+//  Copyright © 2020 Dojo. All rights reserved.
+//
+
+import UIKit
+
+class SearchResultController {
+    
+    let baseURL = URL(string: "https://itunes.apple.com/search")!
+//    private lazy var searchURL = URL(string: "?term=\(searchTerm.self)&entity=\(ResultType.self)", relativeTo: baseURL)
+    var searchResults: [SearchResult] = []
+    
+    func performSearch(searchTerm: String, resultType: ResultType, completion: @escaping (Error?) -> Void) {
+        print(searchTerm)
+        print(resultType)
+        var urlComponents = URLComponents(url: baseURL, resolvingAgainstBaseURL: true)
+        let searchTermQueryItem = URLQueryItem(name: "term", value: searchTerm)
+        let resultTermQueryItem = URLQueryItem(name: "entity", value: resultType.rawValue)
+        urlComponents?.queryItems = [searchTermQueryItem, resultTermQueryItem]
+        print(urlComponents?.url! ?? nil!)
+        print()
```

Comment:
I really appreciate the effort you put in debugging. I know it can be frustrating at times, but I feel like you grow when you really dig in and try to figure out why things are breaking.